### ISSUE TITLE
FixMatch accuracy on CIFAR10-LT

### ISSUE THREAD

**Issue by YUE-FAN** [ytaek-oh/daso#4]
Hi, 

Thanks for the great work. I had a concern regarding the performance of FixMatch on CIFAR-10-LT in Table 1. With N1=1500, M1=3000, DARP reported 71.5% and 68.5% test accuracy for r=100 and 150, respectively. But your numbers are 77.5% and 72.4%, respectively. Could you please explain what is causing the performance difference here? 

Thanks!

**Comment by mortezamg63** [comment]
The same here

**Comment by ytaek-oh** [comment]
Thank you for your interest in our research!
It is worth noting that there are several implementation differences between our codebase and the official DARP implementation. 

We followed the implementation of the official FixMatch repository and the one from kekmodel (https://github.com/kekmodel/FixMatch-pytorch), which makes strong baseline in imbalanced setups.
(One exception is that we do not apply lr scheduler for better performance in imbalanced setup.)
So, we maintain the identical training setup for all the considered baselines in our paper, while maintaining total training steps to 250k.

In more detail, our process involves concatenating all the labeled data, weak and strong augmentations to the unlabeled data, and then feeding the combined data into the model. Subsequently, we compute pseudo-labels based on the model's output on the same batch.

On the other hand, in the official DARP codebase, pseudo-labels are computed separately. 
In DARP's codebase, 64 unlabeled examples are initially sampled, and one weak and two strong augmentations are subsequently applied, resulting in an effective batch size of 64. 
In contrast, we sample 128 unlabeled examples (i.e., \mu=2) and apply weak and strong augmentations once separately.

Another difference is that we do not include labeled data in the unlabeled dataset in case of FixMatch, whereas DARP does. 

Lastly, it's worth noting that DARP utilizes the Adam optimizer, following official ReMixMatch implementation (in my opinion), while we used SGD with momentum.

For a more detailed examination of these variations, I recommend reviewing the codes in the official DARP repository, specifically for `dataset/all_cifar10.py` and `./training_functions.py`.
